### ISSUE TITLE
How to keep this project active...

### ISSUE THREAD

**Issue by rib** [jni-rs/jni-rs#347]
Hi,

Having a look at some of the outstanding issues and pull requests it's easy to see that this project isn't currently being actively developed or maintained and I'm wondering what can be done to potentially change this...

In particular I see this project as a useful building block for building Rust applications on Android and it seems like it'd be best if it's possible to revive it here instead of the current situation where there are multiple independent forks being maintained.

I just did a quick survey of out-standing pull requests and there are numerous PRs that seem to be approved, some of which look like fairly trivial, uncontroversial changes and it's not really clear what's blocking them from being merged. 

E.g.
https://github.com/jni-rs/jni-rs/pull/337 - Fix header generation documentation (trivial documentation fix) (2 months old)
https://github.com/jni-rs/jni-rs/pull/339 - Fix build.rs compilation on android (trivial fix, also supersedes #317) (2 months old)
https://github.com/jni-rs/jni-rs/pull/320 - safety: Mark new_direct_byte_buffer as unsafe (12 months old)
https://github.com/jni-rs/jni-rs/pull/319 - Fix #301: Relax new_local_ref lifetime restrictions (13 months old)
https://github.com/jni-rs/jni-rs/pull/318 - Don't tie the lifetime of AutoArray to a particular JNIEnv ref (13 months old)
https://github.com/jni-rs/jni-rs/pull/308 - Remove empty test (trivial, approved and not waiting for review) (14 months old)

@stanislav-tkach it looks like you're currently the only developer that's keeping the lights on here providing some review for patches but also not merging many changes. In particular I see that you're routinely requesting a second review from @dmitry-timofeev who doesn't seem to have been actively working on jni-rs since 2019, except for handling releases (just based on skimming the commits).

I'm guessing that the original maintainers for this project (such as @dmitry-timofeev) no longer have time to do reviews for this project any more and maybe there's some way to try and recruit others that may be able to at least help review pull requests and try and keep things moving here?

_I also just saw that @dmitry-timofeev lives in Kiev too, so I guess there may be more important concerns than jni-rs for them currently :(_

Is it perhaps possible to add some additional maintainers that are willing to review pull requests? I can volunteer to try and review / ACK some of the outstanding PRs if that might help? Maybe other more-recent contributors can be pinged to see if they'd be willing to review other changes, or perhaps accept that not all changes will get reviewed by multiple people?

**Comment by stanislav-tkach** [comment]
You are right, unfortunately current maintainers don't have time and/or willpower to actively participate in the project development. I would gladly add new people to the `jni-rs` organization, but the only person with the "owner" permissions is @jrobsonchase.

**Comment by dmitry-timofeev** [comment]
Hey, 

I support extending the list of maintainers, as, unfortunately, I can't dedicate time to this project in my current role.

@stanislav-tkach , whilst only Josh can add people to the org, it looks like we can add "Outside collaborators" to this repository with write access. Would that work?

I also suggest reaching out to the recent contributors to see if for anyone it is an important dependency, and they are willing to step up. In my previous role, we used jni-rs quite extensively in one of the projects, therefore my team members could dedicate work time for maintenance.

**Comment by stanislav-tkach** [comment]
@dmitry-timofeev Yes, I think that write access is enough to approve and merge pull request.

**Comment by dmitry-timofeev** [comment]
Great, thanks! 

Then we need to 
1. Find contributors who can and are willing to take these responsibilities. Depending on whether Stas can review CLs, we might need at least two.
2. (if needed) Clarify how changes are integrated (I think there was a thread in the org on the topic)
3. Figure out how to release new versions.

Who could lead this conversation?

**Comment by maurolacy** [comment]
I can help maintaining this.

**Comment by jrobsonchase** [comment]
Hey!

Sorry about my long silence and poor org-ownership. Changed jobs a couple of times since the org was created and it completely fell off my radar.

I've made both @dmitry-timofeev and @stanislav-tkach owners. Y'all can update permissions/membership as you see fit :slightly_smiling_face:

**Comment by rib** [comment]
hi @stanislav-tkach + @dmitry-timofeev  I may also be able to help a bit with maintenance here

I'm working a lot on improving general support for native Rust applications on Android with: https://github.com/rib/android-activity and familiar with JNI lower levels in general.

some time ago I found I needed to make changes to jni-rs to build this Bluetooth UI application for Rust: https://github.com/rib/bluey/tree/main/bluey-ui and currently maintain this custom branch while this project is stalled: https://github.com/rib/jni-rs/commits/master

When I originally opened this issue I did an initial survey of some of the outstanding, credible-looking, work that looks like it needs attention (as itemized above) and I'd be interested in following up on that.

regards

**Comment by stanislav-tkach** [comment]
I don't want to do rushed decisions, but I also don't want for this library to die because of the lack of activity. 

@rib I have invited you to the `jni-rs` team. Personally I would like to recheck the "recently" merged pull request as well as the existing issues and see if a new version should be released, but it is up to you now. 🙃

**Comment by rib** [comment]
Okey, great, thanks. I'll try to find time to look at this soon. Thanks for adding me @stanislav-tkach 

When you say you'd like to recheck recently merged PRs is that because you're concerned about some work that has landed that you weren't confident about?

At the very least the changes since the last release warrant a review to update the change log if not done already so hopefully we can spot something unexpected here.

Since I've been getting away with being based on master for a while I'm optimistic that the outstanding changes in master will be ok to release.

It'd probably be best to try and spin a release sooner rather than later if possible (even if we dont manage to land all the outstanding PRs that may need further work etc) to hopefully signal that the project is still alive.

**Comment by rib** [comment]
> Personally I would like to recheck the "recently" merged pull request

Just looking at the commits since 0.19 it looks like the "Unreleased" info in the changelog is accurate and there are currently only two minor changes (one api marked unstable and jobobject wrappers now implement Default as null)

other changes just affect things like CI

**Comment by rib** [comment]
For reference here, I've created a [0.20 milestone](https://github.com/jni-rs/jni-rs/milestone/2) to help track which PRs seem like they are mostly ready to land as they are. Some issues are blocked from being merged simply due to unrelated clippy lint errors which I've opened #359 for (I can't currently merge that independently because there is a protected branch rule that requires all PRs to be separately reviewed)

**Comment by TobiasDeBruijn** [comment]
If still needed, I'd be more than happy to dedicate time to help maintaining this project. Considering I use it a lot, it'd be only fair to help.

**Comment by OxygenCobalt** [comment]
Any updates on this? Project seems even deader than it was (no commits since early 2024 it seems). Considering forking this off if not just to add #562.

**Comment by rib** [comment]
Hi @OxygenCobalt, although it's true that this project hasn't been maintained over the last year, I have recently found a little bit of time to try and pick up a few things and there's a small chance I'll get a bit more time soonish, as the contract I've been busy with is coming to an end.

I think it might make sense to close this particular issue that I created some years ago when the project was also idle.

There's a more recent discussion here essentially asking the same question though, and maybe we can consolidate there: https://github.com/jni-rs/jni-rs/discussions/546
